File: apps/ppp/src/problems/design-patterns/(factory)/payment-systems/python/code.py
from enum import Enum

class PaymentSystemType(Enum):
  PAYPAL = "paypal"
  WEBMONEY = "webmoney"
  CAT_BANK = "cat-bank"







def payment(
  type: PaymentSystemType,
  base: int,
  amount: int
) -> int:
  raise NotImplementedError

File: apps/ppp/src/problems/design-patterns/(factory)/payment-systems/ruby/code.rb
# PaymentSystemType Enum
module PaymentSystemType
  PAYPAL = "paypal"
  WEBMONEY = "webmoney"
  CAT_BANK = "cat-bank"
end

# @param {String} type - One of PaymentSystemType (PAYPAL, WEBMONEY, CAT_BANK)
# @param {Integer} base
# @param {Integer} amount
# @return {Integer}
def payment(type, base, amount)
  raise "Not implemented"
end
